src/vm/datatypes/strings.c

```diff
@@ -668,6 +668,51 @@ static Value isLowerString(DictuVM *vm, int argCount, Value *args) {
     return BOOL_VAL(true);
 }
 
+static Value collapseSpacesString(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 0) {
+        runtimeError(vm, "collapseSpaces() takes no arguments (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    char *string = AS_CSTRING(args[0]);
+
+    int i, j;
+    for (i = j = 0; string[i]; ++i) {
+        if(!isspace(string[i]) || (i > 0 && !isspace(string[i-1]))) {
+            string[j++] = string[i];
```

Comment:
This is unfortunately going to mutate the original string passed in - we'd need to make a copy first.

As a test, run the following code:

```
const x = "hello      test";
x.collapseSpaces();
print(x);
```